### PR TITLE
Fix vagrant networking

### PR DESCRIPTION
**Pull request by rajatchopra**
Include the master as part of overlay. And remove STP for efficiency.
Should fix broken e2e testcases.

@derekwaynecarr Review please.

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
Thanks. Will test in morning.

**Comment by derekwaynecarr** [comment]
Please remove this block here: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/service.go#L39

**Comment by derekwaynecarr** [comment]
Ignore my previous comment on service.go - which appears to prereq skydns.

**Comment by derekwaynecarr** [comment]
I got the following:

```shell
$ make release
$ cluster/kube-up.sh
$ hack/e2e-suite/services.sh
...
Test 1: Prove that the service portal is alive.
Verifying the portals from the host
Connection to 127.0.0.1 closed.
Checking if service-10807-182tv service-10807-jzjpe service-10807-zi4hx  == service-10807-182tv service-10807-jzjpe service-10807-zi4hx 
Connection to 127.0.0.1 closed.
Checking if service-10807-182tv service-10807-jzjpe service-10807-zi4hx  == service-10807-182tv service-10807-jzjpe service-10807-zi4hx 
Connection to 127.0.0.1 closed.
Checking if service-10807-182tv service-10807-jzjpe service-10807-zi4hx  == service-10807-182tv service-10807-jzjpe service-10807-zi4hx 
Connection to 127.0.0.1 closed.
Checking if service-16629-9gvmk service-16629-ir0xc service-16629-ta0op  == service-16629-9gvmk service-16629-ir0xc service-16629-ta0op 
Verifying the portals from a container
Connection to 127.0.0.1 closed.
service-10807 portal failed from container, expected:

        	service-10807-182tv
	service-10807-jzjpe
	service-10807-zi4hx

        got:

        	download
	out
	service-10807-182tv
	service-10807-jzjpe
	service-10807-zi4hx
	timed
	wget:
```

**Comment by derekwaynecarr** [comment]
So I think the use of public ips in that test case borks us in services.sh

On a side note, how did you test this?  Is there a sample pod setup you ran?

**Comment by rajatchopra** [comment]
@derekwaynecarr 
Quite a bit of the services tests was broken for vagrant. Not just the networking.
It runs clean at my end now. Please re-review.

**Comment by rajatchopra** [comment]
btw.. the public ips creates a problem if you re-run the tests. Because the iptables rules just keep getting appended with new entries for the same public-ips. But the old ones prevail.

This is what I ran on the master to cleanup. But hopefully nobody needs them ever :).
```
rcs=$(kubectl get rc | grep -v CONTROLLER | cut -d " " -f 1 | xargs)
for i in $rcs; do kubectl delete rc $i; done
pods=$(kubectl get pods | cut -d " " -f 1 | grep -v POD | xargs)
for i in $pods; do kubectl delete pods $i; done
se=$(kubectl get services | grep "service" | cut -d " " -f 1 | xargs)
for i in $se; do kubectl delete se $i; done
```

**Comment by derekwaynecarr** [comment]
Trying this again now

File: Vagrantfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ END
 end
 
 # The number of minions to provision
-$num_minion = (ENV['NUM_MINIONS'] || 1).to_i
+$num_minion = (ENV['NUM_MINIONS'] || 2).to_i
```
Comment (derekwaynecarr) [inline Vagrantfile]: Keep the default at 1.

File: cluster/vagrant/config-default.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@
 ## Contains configuration values for interacting with the Vagrant cluster
 
 # Number of minions in the cluster
-NUM_MINIONS=${NUM_MINIONS-"1"}
+NUM_MINIONS=${NUM_MINIONS-"2"}
```
Comment (derekwaynecarr) [inline cluster/vagrant/config-default.sh]: Keep default 1

**Comment by derekwaynecarr** [comment]
Verified the services.sh completes.

Please update the PR to ensure that the kubelet is not reported as running until the sls: sdn module is completed per our discussion.

**Comment by derekwaynecarr** [comment]
Verified that ping works from pod to public internet as well.

**Comment by rajatchopra** [comment]
sls: sdn dependency added to kubelet/init.sls. Thanks Derek.

**Comment by derekwaynecarr** [comment]
LGTM - @smarterclayton can you merge this please.

**Comment by derekwaynecarr** [comment]
Fixed #3946

File: cluster/saltbase/salt/top.sls [hunk 3]
```diff
@@ -40,6 +40,9 @@ base:
     - kube-addons
 {% if grains['cloud'] is defined and grains['cloud'] == 'azure' %}
     - openvpn
+{% else %}
+    - docker
+    - sdn
```
Comment (zmerlynn) [inline cluster/saltbase/salt/top.sls]: I believe this pull broke GCE.

I think this ended up enabling docker and sdn on *everything*, didn't it?

Comment (derekwaynecarr) [inline cluster/saltbase/salt/top.sls]: Ah sorry, it did not enable sdn on everything as that is conditional, but it did put docker on everything.  Apologies, missed that.  @rajatchopra - can we update to not require docker on the kubernetes-master, but just have the sdn bit?

**Comment by derekwaynecarr** [comment]
So if we required docker on the master to have the tunnel to include in the overlay, then we need to modify sdn/init.sls to require the docker module so it only affects vagrant.  Or we can modify the provision-network.sh script to work even if docker is not installed.

**Pull request merged**